ext/net/ops.rs

```diff
@@ -59,6 +63,17 @@ pub struct IpAddr {
   pub port: u16,
 }
 
+fn process_ip_addr(addr: &mut IpAddr) -> Result<(), AnyError> {
+  let extracted_host = extract_host(&addr.hostname);
+  let tmp_host = &extracted_host.clone();
+  let (host_str, port) = split_host_port(&extracted_host)?;
+  addr.hostname = Host::from_str(&host_str, &tmp_host)?.to_string();
+  if let Some(port) = port {
+    addr.port = port;
+  }
+  Ok(())
+}
```

Comment:
What is the purpose of this function? A docstring here would be very useful, also if it's important there should be a unit test for it. I'm also not keen on having a mutable reference here if it's a validation function that doesn't return a value.

Reply:
It is an  handle for host before sent it to
01_net.js 

When I try to do 

Deno.connect({ hostname: "https://192.0.2.1/", port: 80 });
   Deno.listen({ hostname: "https://golang.org/", port: 80, transport: "tcp" });

It will send the hostname and port as args 
to
   async function call(media) {
function listen(media) {
In ext/net/01_net.js

So it will try to connect to invalid addresses, so we need to deal with the host and extract the valid port and host before connecting